docs/getting-started/concepts/dataset.md

```diff
@@ -0,0 +1,43 @@
+# Dataset
+
+Feast datasets allow for conveniently saving dataframes that include both features and entities to be subsequently used for data analysis and model training.
+[Data Quality Monitoring](https://docs.google.com/document/d/110F72d4NTv80p35wDSONxhhPBqWRwbZXG4f9mNEMd98) was the primary motivation for creating dataset concept.
+
+Dataset's metadata is stored in the Feast registry and raw data (features, entities, additional input keys and timestamp) is stored in the [offline store](../architecture-and-components/offline-store.md).
+
+Dataset can be created from:
+1. Results of historical retrieval
+2. [planned] Logging request (including input for [on demand transformation](../../reference/alpha-on-demand-feature-view.md)) and response during feature serving
+3. [planned] Logging features during writing to online store (from batch source or stream)
+
+
+### Creating Saved Dataset from Historical Retrieval
+
+To create a saved dataset from historical features for later retrieval or analysis, use the `save_as` parameter of `get_historical_features` method.
+If this parameter is supplied, Feast eagerly computes the dataframe and stores the data using specified `storage`.
+Any offline store can be used as such storage. Out of the box Feast supports BigQuery, Redshift and File storage (local, Google Storage or S3).
+
+```python
+
+from feast import FeatureStore
+from feast.saved_dataset import SavedDatasetOptions
+from feast.infra.offline_stores.bigquery_source import SavedDatasetBigQueryStorage
+
+store = FeatureStore()
+
+store.get_historical_features(
+    features=["driver:avg_trip"],
+    entity_df=...,
+    save_as=SavedDatasetOptions(
+        name='my_training_dataset',
+        storage=SavedDatasetBigQueryStorage(table_ref='<gcp-project>.<gcp-dataset>.my_training_dataset'),
+        tags={'author': 'oleksii'}
+    )
+)
+```
+
+Saved dataset can be later retrieved using `get_saved_dataset` method:
+```python
+retrieval_job = store.get_saved_dataset('my_training_dataset')
```

Comment:
Is this when the saved dataset is computed ?

Reply:
If by compute you mean materialized, then no. The saved dataset is computed eagerly on creation. For example, when you call `get_historical_features` with the `save_as` parameter saved dataset would be already materialized & stored to offline store & registry by the time this function returns.

Reply:
> For example, when you call get_historical_features with the save_as parameter saved dataset would be already materialized

That's where it could be confusing because `fs.get_historical_features()` does not materialize anything normally but it woudl do it when `save_as` is passed. Two completely different behaviour with the same method.